server/utils/util.go

```diff
@@ -0,0 +1,119 @@
+package utils
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+var l = log.New(os.Stderr, "", 1)
+
+// Logs error to Stderr.
+func LogError(a ...interface{}) {
+	l.Println(a...)
+}
+
+// Generates Random string using crypto/rand.
+func CryptoRandomString(len int) (string, error) {
+	random := make([]byte, len)
+	_, err := rand.Reader.Read(random)
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(random), nil
+}
+
+// Enables content security policy which only allows self scripts as well as specific nonces for inline scripts.
+func EnableContentSecurityPolicy(w *http.ResponseWriter) {
+	ContentSecurityPolicyString := strings.Join([]string{
+		"script-src 'self' 'nonce-EDNnf03nceIOfn39fn3e9h3sdfa' 'nonce-afds3h9e3nf93nf0Iecn30fnNDE'",
+		"style-src 'self' 'unsafe-inline' fonts.googleapis.com ",
+		"object-src 'none'"}, ";")
+	(*w).Header().Set("Content-Security-Policy", ContentSecurityPolicyString)
+}
+
+// Gets header from request based on key.
+func GetHeader(r *http.Request, key string) string {
+	return r.Header.Get(key)
+}
+
+// Extracts a token from a cookie based on token name.
+func ExtractTokenFromCookie(tokenName string, r *http.Request) (string, error) {
+	cookie, err := r.Cookie(tokenName)
+	if err != nil {
+			return "", errors.New(tokenName + " not found")
+	}
+	return cookie.Value, nil
+}
+
+// Saves a token cookie with httpOnly and Secure flag.
+func SaveCookie(cookieName, token string, w http.ResponseWriter) {
+	cookie := &http.Cookie{
+		Name:     cookieName,
+		Value:    token,
+		MaxAge:   0,
+		SameSite: http.SameSiteNoneMode,
+		HttpOnly: true,
+		Secure:   true,
+		Path:     "/",
+	}
+	http.SetCookie(w, cookie)
+	w.Header().Add("Vary", "Cookie")
```

Comment:
What's the purpose of the `Vary` header?

Reply:
I now see that that error indentation is 3 tabs, I will fix that. My understanding from the formatting document is to have 2 tabs (gofmt makes it 1 tab ?). Please clarify. Thank you!

I have removed vary, it was unneeded I believe.

Reply:
You can just stick with what `gofmt` does.

Reply:
ok I will run gofmt. :)